python/cudf/cudf/core/frame.py

```diff
@@ -473,7 +473,8 @@ def _gather(self, gather_map, keep_index=True):
                 self, as_column(gather_map), keep_index=keep_index
             )
         )
-        result._copy_categories(self)
+        result._postprocess_columns(self)
+        result._copy_struct_names(self)
```

Comment:
Should this be combined into `_postprocess_columns`?

Reply:
It is, just called copy_struct_names redundantly here. Removed